googletest/include/gtest/gtest.h

```diff
@@ -1992,11 +1992,11 @@ class TestWithParam : public Test, public WithParamInterface<T> {
 // Note:
 //
 //   1. It is possible to make a user-defined type work with
-//   {ASSERT|EXPECT}_??(), but that requires overloading the
-//   comparison operators and is thus discouraged by the Google C++
-//   Usage Guide.  Therefore, you are advised to use the
-//   {ASSERT|EXPECT}_TRUE() macro to assert that two objects are
-//   equal.
+//   {ASSERT|EXPECT}_??() by overloading the comparison operators. Note
+//   that these macros rely on argument-dependent lookup to find the
+//   comparison operators, so the operators must be defined in the same
+//   namespace as the types being compared (or at least one of the
+//   types if they are from different namespaces).
```

Comment:
You are correct that the style guide doesn't discourage this anymore, so a change here is in order.

However, the note about ADL isn't correct. The operator doesn't need to be in the same namespace (though best-practices would suggest it should be). https://godbolt.org/z/sPTxzn.

Reply:
> The operator doesn't need to be in the same namespace (though best-practices would suggest it should be). 

Sorry, my phrasing wasn't clear -- I meant that `EXPECT_EQ` relies on ADL to find the comparison operator, not that ADL is necessary in the general case. Using the code sample you linked, notice that even though `a == b` compiles fine, `EXPECT_EQ(a, b)` does not: https://godbolt.org/z/nTnhWo. I can take a closer look at the implementation of `EXPECT_EQ`, though.